src/Microsoft.DotNet.Tools.Compiler.Native/ArgValues.cs

```diff
@@ -56,8 +57,20 @@ public NativeCompileSettings GetNativeCompileSettings()
 
             if (!string.IsNullOrEmpty(IlcPath))
             {
-                config.IlcPath = IlcPath;
-                config.IlcSdkPath = IlcPath;
+                // We want a directory path. If the user gave us the exact path to the executable
```

Comment:
I try to avoid properties that doesn't round trip. I generally expect that any property passes this test:
Foo.Prop = x
Assert(Foo.Prop == x);

If you feel strongly that it should be NativeCompileSettings I would propose changing the property setter to a method to better demonstrate that it has side effects. However unless the entire ArgValues type is going away I'd still contend that doing this mutation during the conversion of ArgValues to NativeCompileSettings is the better place for it, or potentially even earlier during parsing. By the time the data reaches NativeCompileSettings, I think of that as the model in a MVC or MVVM paradigm. The model should not have to be concerned with ambiguities of input.

Just let me know what you'd like though. In the end we may not agree but I'm just a drop-in contributor and I assume you are maintaining this code so you get the final say : )